### PR TITLE
chore: v9 upgrade handler

### PR DESCRIPTION
**Pull request by glnro**
This PR fulfills the acceptance criteria of #2038 for the #2034 epic.

Changes include
- Updating gaia module to v9
- Upgrade handler for v8 -> v9
- Updating the upgrade workflow test to model v8 -> v9 upgrade
- Refactors for ICS integration
- Update of ICS dependency to current main
- E2E gov tests for adding and removing consumer chains

### CODE REVIEW

File: contrib/scripts/run-upgrade-commands.sh [hunk 1]
```diff
@@ -71,5 +71,5 @@ if test -f "$BINARY"; then
   echo "Done \n"
 
 else
-  echo "Please build gaia v7 and move to ./build/gaiad7"
+  echo "Please build gaia v8 and move to ./build/gaiad8"
```
Comment (MSalopek) [inline contrib/scripts/run-upgrade-commands.sh:74]: I would extend this message to:
`Target gaia binary could not be found. Please build gaia v8 and move to ./build/gaiad8`

It was a bit unclear to me what it was the purpose of it.

File: tests/e2e/e2e_gov_test.go [hunk 2]
```diff
@@ -124,6 +127,65 @@ func (s *IntegrationTestSuite) GovCommunityPoolSpend() {
...
+	s.execQueryConsumerChains(s.chainA, 0, gaiaHomePath, validateConsumerAddition, consumerChainID)
+
+	// Remove Consumer Chain
+	proposalCounter++
+	submitGovFlags = []string{"consumer-removal", configFile(proposalRemoveConsumerChainFilename)}
+	depositGovFlags = []string{strconv.Itoa(proposalCounter), depositAmount.String()}
+	voteGovFlags = []string{strconv.Itoa(proposalCounter), "yes"}
+	s.runGovProcess(chainAAPIEndpoint, sender, proposalCounter, ccvtypes.ProposalTypeConsumerRemoval, submitGovFlags, depositGovFlags, voteGovFlags, "vote")
+
+	// Query and ssert consumer has been removed
```
Comment (sainoe) [inline tests/e2e/e2e_gov_test.go]:
```suggestion
	// Query and assert consumer has been removed
```

**Review by sainoe — APPROVED** [review]
LGTM. Nice work!

File: tests/e2e/e2e_query_exec_test.go [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+	"context"
+	"fmt"
+	"time"
+
+	ccvtypes "github.com/cosmos/interchain-security/x/ccv/provider/types"
+
+	"github.com/cosmos/cosmos-sdk/client/flags"
+)
+
+func (s *IntegrationTestSuite) execQueryConsumerChains(
```
Comment (MSalopek) [inline tests/e2e/e2e_query_exec_test.go:13]: Maybe also check that proper clients and channels got established.
The more complex details are checked in ICS.

**Review by MSalopek — APPROVED** [review]
@glnro Added some nits. It would be great to test that clients and channels are created correctly when adding consumer chains.

File: tests/e2e/e2e_query_exec_test.go (hunk 3 above)
Comment (glnro) [reply tests/e2e/e2e_query_exec_test.go:13]: Yeah great idea I'll add in a follow up PR

**Review by mpoke — APPROVED** [review]
Approving based on the review of @sainoe and @MSalopek

**Pull request merged**